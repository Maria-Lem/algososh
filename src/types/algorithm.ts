export enum Algorithm {
  SelectionSort = "selection sort",
  BubbleSort = "bubble sort",
}