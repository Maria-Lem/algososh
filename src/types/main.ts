import { ElementStates } from "./element-states";

export interface IIsLoader {
  [name: string]: boolean
}

export interface IResultArray {
  value: string;
  state: ElementStates;
}

export interface IRandomArray {
  value: number;
  state: ElementStates;
}