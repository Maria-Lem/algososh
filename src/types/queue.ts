import { ElementStates } from "./element-states";

export interface IQueueElement {
  item: string;
  state: ElementStates;
}

export interface IQueue<T> {
  enqueue: (item: T) => void;
  dequeue: () => void;
  peek: () => T | IQueueElement | null;
  getElements: () => T[] | IQueueElement[];
  clear: () => void;
  getHead: () => number;
  getTail: () => number;
  getSize: () => number;
  isEmpty: () => boolean;
}