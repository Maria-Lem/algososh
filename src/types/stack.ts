export interface IStack<T> {
  push: (item: T) => void;
  pop: () => void;
  peek: () => T | null;
  getElements: () => T[];
  clear: () => void;
}