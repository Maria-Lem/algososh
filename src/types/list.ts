import { LinkedListNode } from "../components/list-page/list-class";
import { IResultArray } from "./main";

export interface ILinkedListElement extends IResultArray {
  isSmallCircleTop?: boolean;
  isSmallCircleBottom?: boolean;
}

export interface ILinkedList<T> {
  addToHead: (item: T) => void;
  addToTail: (item: T) => void;
  removeFromHead: () => void;
  removeFromTail: () => void;
  insertAtIndex: (item: T, index: number) => void;
  removeByIndex: (index: number) => void;
  printArrayFromLinkedList: () => T[];
  getSize: () => void;
  getHead: () => LinkedListNode<T> | null;
  getTail: () => LinkedListNode<T> | null;
}