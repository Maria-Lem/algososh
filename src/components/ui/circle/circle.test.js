import renderer from 'react-test-renderer';
import { Circle } from "./circle";
import { ElementStates } from '../../../types/element-states';

describe('Testing circle functionality', () => {
  it('Circle should render correctly without a letter', () => {
    const circle = renderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should be rendered correctly with letters in it', () => {
    const circle = renderer.create(<Circle letter='1213' />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render a head element with a letter in it', () => {
    const circle = renderer.create(<Circle head='A' />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render a head element with an element in it', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render a tail element with a letter in it', () => {
    const circle = renderer.create(<Circle tail='Z' />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render a tail element with an element in it', () => {
    const circle = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render index', () => {
    const circle = renderer.create(<Circle index={3} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render with props isSmall = true', () => {
    const circle = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render default state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render changing state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it('Circle should correctly render modified state', () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(circle).toMatchSnapshot();
  });
});