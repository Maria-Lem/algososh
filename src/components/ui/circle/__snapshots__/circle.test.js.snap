// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing circle functionality Circle should be rendered correctly with letters in it 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      1213
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render a head element with a letter in it 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    A
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render a head element with an element in it 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render a tail element with a letter in it 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Z
  </div>
</div>
`;

exports[`Testing circle functionality Circle should correctly render a tail element with an element in it 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Testing circle functionality Circle should correctly render changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    3
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should correctly render with props isSmall = true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle functionality Circle should render correctly without a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
